apps/vocabulary/main.go

```diff
@@ -0,0 +1,173 @@
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/googleapis/gnostic/openapiv2"
+)
+
+func readDocumentFromFileWithName(filename string) (*pb.Document, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	document := &pb.Document{}
+	err = proto.Unmarshal(data, document)
+	if err != nil {
+		return nil, err
+	}
+	return document, nil
+}
+func addToMap(word string, mapName map[string]int) {
+	mapName[word] += 1
+}
+
+func processDocument(document *pb.Document, schemas map[string]int, operationId map[string]int, names map[string]int, properties map[string]int) {
+	if document.Definitions != nil {
+		for _, pair := range document.Definitions.AdditionalProperties {
+			addToMap(pair.Name, schemas)
+			processSchema(pair.Value, properties)
+		}
+	}
+	for _, pair := range document.Paths.Path {
+		v := pair.Value
+		if v.Get != nil {
+			processOperation(v.Get, operationId, names)
+		}
+		if v.Post != nil {
+			processOperation(v.Post, operationId, names)
+		}
+		if v.Put != nil {
+			processOperation(v.Put, operationId, names)
+		}
+		if v.Patch != nil {
+			processOperation(v.Patch, operationId, names)
+		}
+		if v.Delete != nil {
+			processOperation(v.Delete, operationId, names)
+		}
+	}
+}
+
+func processOperation(operation *pb.Operation, operationId map[string]int, names map[string]int) {
+	if operation.OperationId != "" {
+		addToMap(operation.OperationId, operationId)
+	}
+	for _, item := range operation.Parameters {
+		switch t := item.Oneof.(type) {
+		case *pb.ParametersItem_Parameter:
+			switch t2 := t.Parameter.Oneof.(type) {
+			case *pb.Parameter_BodyParameter:
+				addToMap(t2.BodyParameter.Name, names)
+			case *pb.Parameter_NonBodyParameter:
+				switch t3 := t2.NonBodyParameter.Oneof.(type) {
+				case *pb.NonBodyParameter_FormDataParameterSubSchema:
+					addToMap(t3.FormDataParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_HeaderParameterSubSchema:
+					addToMap(t3.HeaderParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_PathParameterSubSchema:
+					addToMap(t3.PathParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_QueryParameterSubSchema:
+					addToMap(t3.QueryParameterSubSchema.Name, names)
+				}
+			}
+		}
+	}
+}
+
+func processSchema(schema *pb.Schema, properties map[string]int) {
+	if schema.Properties == nil {
+		return
+	}
+	for _, pair := range schema.Properties.AdditionalProperties {
+		addToMap(pair.Name, properties)
+	}
+}
+
+func main() {
+	flag.Parse()
```

Comment:
consider pulling some of this method out into separate methods - this one is getting a little too long